qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDbConnector.groovy

```diff
@@ -26,52 +29,27 @@ import life.qbic.portal.portlet.exceptions.DatabaseQueryException
 @Log4j2
 class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDataSource, SearchCustomerDataSource, CreateAffiliationDataSource, ListAffiliationsDataSource {
 
+
+  @Deprecated
   CustomerDatabaseQueries databaseQueries
+  private final Connection connection
+
+  private static final String CUSTOMER_SELECT_QUERY = "SELECT id, first_name AS firstName, last_name AS lastName, title as academicTitle, email as eMailAddress FROM customer"
+  private static final String AFFILIATION_SELECT_QUERY = "SELECT id, organization AS organisation, address_addition AS addressAddition, street, postal_code AS postalCode, city, country, category FROM affiliation"
 
-  CustomerDbConnector(CustomerDatabaseQueries databaseQueries){
+  @Deprecated
+  CustomerDbConnector(CustomerDatabaseQueries databaseQueries, Connection connection){
     this.databaseQueries = databaseQueries
+    this.connection = connection
   }
 
-  @Override
-  List<Customer> findCustomer(String firstName, String lastName) throws DatabaseQueryException {
-    throw new Exception("Method not implemented.")
+  CustomerDbConnector(Connection connection) {
+    this.connection = connection
   }
 
-  /**
-   * @inheritDoc
-   * @param criteria
-   * @return
-   */
   @Override
-  List<Customer> findCustomer(SearchCriteria criteria) {
-
-    List<Customer> foundCustomers = null
-
-    criteria.criteria.each { criteriaKey, criteriaValue ->
-
-      switch(criteriaKey){
-        case CriteriaType.FIRST_NAME:
-          foundCustomers.addAll(databaseQueries.findPersonByName(criteriaValue))
-          break
-        case CriteriaType.LAST_NAME:
-          foundCustomers.addAll(databaseQueries.findPersonByName(criteriaValue))
-          break
-        case CriteriaType.GROUP_NAME:
-          foundCustomers.addAll(databaseQueries.findCustomerByGroup(criteriaValue))
-          break
-        case CriteriaType.ADD_ADDRESS:
-          foundCustomers.addAll(databaseQueries.findCustomerByAdditionalAddress(criteriaValue))
-          break
-        case CriteriaType.CITY:
-          foundCustomers.addAll(databaseQueries.findCustomerByCity(criteriaValue))
-          break
-        default:
-          break
-      }
-    }
-
-    //todo filter the found customers for customers that fulfill all searchcriteria!
-    return foundCustomers
+  List<Customer> findCustomer(String firstName, String lastName) throws DatabaseQueryException {
+    throw new Exception("Method not implemented.")
```

Comment:
Are you planning to implement this? I think because it worked before I think you can go ahead and provide this method too.

Reply:
Yes that is what my card is about. I would like to implement it without a switch case logic and declutter it a bit. I also do believe that we do not need the SearchCriteria method any longer and removed the SearchCriteria thing for now since we need to think a bit about how the filtering is done.